CMakeLists.txt

```diff
@@ -1336,74 +1333,72 @@ IF(ALSOFT_UTILS)
-	utils/sofa/SOFANcFile.cpp
-	utils/sofa/SOFAPoint3.cpp
-	utils/sofa/SOFAPosition.cpp
-	utils/sofa/SOFAReceiver.cpp
-	utils/sofa/SOFASimpleFreeFieldHRIR.cpp
-	utils/sofa/SOFASimpleFreeFieldSOS.cpp
-	utils/sofa/SOFASimpleHeadphoneIR.cpp
-	utils/sofa/SOFAGeneralTF.cpp
-	utils/sofa/SOFAGeneralFIR.cpp
-	utils/sofa/SOFASource.cpp
-	utils/sofa/SOFAString.cpp
-	utils/sofa/SOFAUnits.cpp
-	utils/sofa/SOFAtoOpenAL.cpp
-   )
-        	SET(HAVE_HRTF_SOFA 1)
-
-		set_source_files_properties(utils/sofa/SOFAAPI.cpp
-	utils/sofa/SOFAAttributes.cpp
-	utils/sofa/SOFACoordinates.cpp
-	utils/sofa/SOFADate.cpp
-	utils/sofa/SOFAEmitter.cpp
-	utils/sofa/SOFAExceptions.cpp
-	utils/sofa/SOFAFile.cpp
-	utils/sofa/SOFAHelper.cpp
-	utils/sofa/SOFAListener.cpp
-	utils/sofa/SOFANcFile.cpp
-	utils/sofa/SOFAPoint3.cpp
-	utils/sofa/SOFAPosition.cpp
-	utils/sofa/SOFAReceiver.cpp
-	utils/sofa/SOFASimpleFreeFieldHRIR.cpp
-	utils/sofa/SOFASimpleFreeFieldSOS.cpp
-	utils/sofa/SOFASimpleHeadphoneIR.cpp
-	utils/sofa/SOFAGeneralTF.cpp
-	utils/sofa/SOFAGeneralFIR.cpp
-	utils/sofa/SOFASource.cpp
-	utils/sofa/SOFAString.cpp
-	utils/sofa/SOFAUnits.cpp
-	utils/sofa/SOFAtoOpenAL.cpp
-	PROPERTIES COMPILE_FLAGS " -DLINUX=1 -D__unix__=1 -DHOST_ENDIAN_LITTLE=1 -Wno-ignored-qualifiers")
+		    utils/sofa/SOFAAPI.cpp
+		    utils/sofa/SOFAAttributes.cpp
+		    utils/sofa/SOFACoordinates.cpp
+		    utils/sofa/SOFADate.cpp
+		    utils/sofa/SOFAEmitter.cpp
+		    utils/sofa/SOFAExceptions.cpp
+		    utils/sofa/SOFAFile.cpp
+		    utils/sofa/SOFAHelper.cpp
+		    utils/sofa/SOFAListener.cpp
+		    utils/sofa/SOFANcFile.cpp
+		    utils/sofa/SOFAPoint3.cpp
+		    utils/sofa/SOFAPosition.cpp
+		    utils/sofa/SOFAReceiver.cpp
+		    utils/sofa/SOFASimpleFreeFieldHRIR.cpp
+		    utils/sofa/SOFASimpleFreeFieldSOS.cpp
+		    utils/sofa/SOFASimpleHeadphoneIR.cpp
+		    utils/sofa/SOFAGeneralTF.cpp
+		    utils/sofa/SOFAGeneralFIR.cpp
+	 	    utils/sofa/SOFASource.cpp
+  		    utils/sofa/SOFAString.cpp
+		    utils/sofa/SOFAUnits.cpp
+		    utils/sofa/SOFAtoOpenAL.cpp)
+        SET(HAVE_HRTF_SOFA 1)
+
+	    set_source_files_properties(utils/sofa/SOFAAPI.cpp
+		    utils/sofa/SOFAAttributes.cpp
+    		utils/sofa/SOFACoordinates.cpp
+	    	utils/sofa/SOFADate.cpp
+	    	utils/sofa/SOFAEmitter.cpp
+	    	utils/sofa/SOFAExceptions.cpp
+	    	utils/sofa/SOFAFile.cpp
+	    	utils/sofa/SOFAHelper.cpp
+	    	utils/sofa/SOFAListener.cpp
+	    	utils/sofa/SOFANcFile.cpp
+	    	utils/sofa/SOFAPoint3.cpp
+	    	utils/sofa/SOFAPosition.cpp
+	    	utils/sofa/SOFAReceiver.cpp
+	    	utils/sofa/SOFASimpleFreeFieldHRIR.cpp
+	    	utils/sofa/SOFASimpleFreeFieldSOS.cpp
+	    	utils/sofa/SOFASimpleHeadphoneIR.cpp
+	    	utils/sofa/SOFAGeneralTF.cpp
+	    	utils/sofa/SOFAGeneralFIR.cpp
+	    	utils/sofa/SOFASource.cpp
+	    	utils/sofa/SOFAString.cpp
+	    	utils/sofa/SOFAUnits.cpp
+	    	utils/sofa/SOFAtoOpenAL.cpp
+	    	PROPERTIES COMPILE_FLAGS " -DLINUX=1 -D__unix__=1 -DHOST_ENDIAN_LITTLE=1 -Wno-ignored-qualifiers")
```

Comment:
Unfortunately, things like this will prevent this from being accepted. It presumes a specific target system, but it should set the needed macros depending on the actual target as reported by cmake (e.g. it should be able to work with big-endian and/or non-Linux system). Consequently, hard-coding linking to ```libnetcdf_c++4.so``` won't work either, since different systems may dress their libs differently.

Also, it shouldn't be cmake's job to download and build netcdfc++, as that's a very system-dependent operation (not all systems will have wget, or can use configure scripts). It should just check for it in the normal way, using a Find* style module that allows specifying a custom location through a ```*DIR``` variable, leaving it to the user to ensure it's already built and set up properly.